src/main/java/org/schemaspy/SchemaAnalyzer.java

```diff
@@ -328,7 +332,14 @@ private void generateHtmlDoc(Config config, boolean useVizJS, ProgressListener p
         progressListener.graphingSummaryProgressed();
 
         boolean showDetailedTables = tables.size() <= config.getMaxDetailedTables();
-        final boolean includeImpliedConstraints = commandLineArguments.withImpliedRelationships();
+
+        // getting implied constraints has a side-effect of associating the parent/child tables, so don't do it
+        // here unless they want that behavior
+        List<ImpliedForeignKeyConstraint> impliedForeignKeyConstraints = impliedConstraint.mutate(tables);
+        boolean hasImpliedConstrains = !impliedForeignKeyConstraints.isEmpty();
```

Comment:
Please clarify if able, preferably with self-documenting code. My first interpretation was that the comment belongs with the command-line parsing above and that we perform the calculations prematurely, but perhaps we are dependent on the side effects?